extensions/refresh-catalog/src/test/java/org/eclipse/dataspaceconnector/mvd/RegistrationServiceNodeDirectoryTest.java

```diff
@@ -45,28 +46,30 @@ void getAll_emptyList() {
 
     @Test
     void getAll() {
-        var service = new RegistrationServiceNodeDirectory(registryApi);
+        var service = new RegistrationServiceNodeDirectory(registryApi, resolver);
 
         var company1 = getParticipant();
         var company2 = getParticipant();
+        FederatedCacheNode node1 = node();
+        FederatedCacheNode node2 = node();
         when(registryApi.listParticipants()).thenReturn(List.of(company1, company2));
+        when(resolver.toFederatedCacheNode(company1)).thenReturn(node1);
+        when(resolver.toFederatedCacheNode(company2)).thenReturn(node2);
 
         List<FederatedCacheNode> cacheNodes = service.getAll();
         assertThat(cacheNodes)
                 .usingRecursiveFieldByFieldElementComparator()
-                .containsExactly(mapToNode(company1), mapToNode(company2));
+                .containsExactly(node1, node2);
     }
 
-    private FederatedCacheNode mapToNode(Participant participant) {
-        return new FederatedCacheNode(participant.getName(), participant.getUrl(), participant.getSupportedProtocols());
+    private FederatedCacheNode node() {
+        return new FederatedCacheNode(faker.lorem().word(), faker.internet().url(), List.of("ids-multipart"));
     }
 
     @NotNull
     private Participant getParticipant() {
         var participant = new Participant();
-        participant.setName(faker.lorem().word());
-        participant.setUrl(faker.internet().url());
-        participant.setSupportedProtocols(List.of(faker.lorem().word(), faker.lorem().word()));
+        participant.setDid(faker.internet().domainName());
```

Comment:
nit: just to stay consistent as we do in the other test.
```suggestion
        participant.setDid(String.format("did:web:%s", faker.internet().domainName()));
```

static import can be used for .format